src/account/mod.rs

```diff
@@ -279,10 +279,21 @@ impl Account {
         from: usize,
         message_type: Option<MessageType>,
     ) -> Vec<&Message> {
+        let mut found_message_payloads = vec![];
         let messages_iter = self
             .messages
             .iter()
+            // iterate in reverse order to find reattachments first
+            .rev()
             .filter(|message| {
+                // ignore if the message if one with the same payload was already found (i.e. this message was reattached)
+                if found_message_payloads
+                    .iter()
+                    .any(|payload| payload == &message.payload())
```

Comment:
@lucasfernog It is possible (though less probable) that an original message gets confirmed even after it was reattached. Considering this, we should always return the confirmed one. So when we find a reattachment, we should check if any of the message is confirmed and then give precedence to the confirmed one.

Reply:
fixed on this [commit](https://github.com/iotaledger/wallet.rs/pull/123/commits/63ebcb2d9b68e283ca945cc5210be06088ff6802)